EmbeddedSocialClient/sdk/src/main/java/com/microsoft/embeddedsocial/server/model/notification/GetNotificationFeedResponse.java

```diff
@@ -16,30 +16,31 @@
 
 public class GetNotificationFeedResponse extends FeedUserResponse implements ListResponse<ActivityView> {
 
-	private List<ActivityView> activities;
-	private String deliveredActivityHandle;
-
-	public GetNotificationFeedResponse(List<ActivityView> activities) {
-		this.activities = activities;
-
-		// FIXME fix setting this handle
-		this.deliveredActivityHandle = "";
-	}
-
-	public GetNotificationFeedResponse (FeedResponseActivityView response) {
-		activities = new ArrayList<>();
-		for (com.microsoft.embeddedsocial.autorest.models.ActivityView view : response.getData()) {
-			activities.add(new ActivityView(view));
-		}
-		this.deliveredActivityHandle = "";
-	}
-
-	@Override
-	public List<ActivityView> getData() {
-		return activities != null ? activities : Collections.emptyList();
-	}
-
-	public String getDeliveredActivityHandle() {
-		return deliveredActivityHandle;
-	}
+    private List<ActivityView> activities;
+    // Setting this handle is required to properly update the "last-read" notification
+    private String deliveredActivityHandle;
+
+    public GetNotificationFeedResponse(List<ActivityView> activities) {
+        this.activities = activities;
+        this.deliveredActivityHandle = (activities.isEmpty() ? "" : activities.get(0).getHandle());
```

Comment:
Can we add a null check for the activities parameter? As is, activities.isEmpty() could throw a null pointer exception.

Reply:
Might want to add @NotNull for activities parameter in method signature too.